[Darren Chang] iP

ChangGittyHub opened the pull request:
# Duuk

> "Be Still and know that I am God" – Psalm 46:10 ([source](https://www.biblegateway.com/passage/?search=Psalm%2046%3A10&version=NIV))

Duuk is Duke's evil twin brother. It's,

- text-based
- fun to learn
- EXTRA FAST to use

All you need to do is,

1. download it from [here](https://www.youtube.com/watch?v=dQw4w9WgXcQ).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is absolutely FREE!

## Features:

- Managing tasks
- Managing deadlines (coming soon)
-  Managing event (coming soon)
-  Tagging (coming soon)

Here's the main method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

ChangGittyHub committed:
Implement Level 1

ChangGittyHub committed:
Implement Level 2

ChangGittyHub committed:
Implement Level 3

ChangGittyHub committed:
Updated Implement Level 3

ChangGittyHub committed:
Implement Level 5

ChangGittyHub committed:
Implement Level 5

ChangGittyHub committed:
Implement Level 6

HmuuMyatMoe commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task {
+    private String dueDate;
+
+    public Deadline(String name, String dueDate) {
```
Suggestion to add java doc comments for all public class and methods.

HmuuMyatMoe commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+import java.util.*;
```
Would be good to list out the imported classes explicitly.

HmuuMyatMoe commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+        this.sc = sc;
+    }
+    public void start(Tasklist list) throws DukeException {
+        while(sc.hasNext()) {
```
```suggestion
        while (sc.hasNext()) {
```

(while should be followed by a white space)

HmuuMyatMoe commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+            input = input.trim();
+            String[] separateInput = input.split(" ");
+            String str = separateInput[0];
+            if(input.equals("bye")) {
```
Perhaps you can consider defining an enum type representing the commands for this part of the code.

HmuuMyatMoe commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,88 @@
+                System.out.println("Bye. Hope to see you again soon!");
+                sc.close();
+                return;
+            } else if(input.equals("list")) {
```
```suggestion
            } else if (input.equals("list")) {
```
'if' should be followed by a whitespace, do remember to add a whitespace for all the other 'if' and 'else if' too.

HmuuMyatMoe commented on src/main/java/Tasklist.java:
```diff
@@ -0,0 +1,40 @@
+        this.list.add(task);
+        System.out.println("Got it. I've added this task:\n " + task.toString());
+    }
+    public void markTaskAsDone(int index) {
```
Great job on the line wrapping here, the code looks neat!

HmuuMyatMoe reviewed (commented):
Code looks neat! Just some modifications are required.

ChangGittyHub committed:
Implement Level-7

ChangGittyHub committed:
Implement Level-8

ChangGittyHub committed:
A -MoreOOP

ChangGittyHub committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

ChangGittyHub committed:
Merge remote-tracking branch 'upstream/add-gradle-support' into branch-Level-8
Merge to fix bug Level 8

ChangGittyHub committed:
Implement Level 8

ChangGittyHub committed:
Merge branch 'branch-Level-8'

ChangGittyHub committed:
Add J-Unit

ChangGittyHub committed:
Create a JAR file

ChangGittyHub committed:
Add JavaDoc comments

ChangGittyHub committed:
Tweak the code to comply with a coding standard

ChangGittyHub committed:
Implement Level-9

ChangGittyHub committed:
Merge remote-tracking branch 'origin/branch-A-CodingStandard'

# Conflicts:
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Storage.java
#	src/main/java/duke/Ui.java
#	src/main/java/exception/DukeException.java
#	src/main/java/task/Event.java
#	src/main/java/task/Tasklist.java
#	src/main/java/task/Todo.java

ChangGittyHub committed:
Merge remote-tracking branch 'origin/branch-Level-9'

# Conflicts:
#	src/main/java/task/Task.java

ChangGittyHub committed:
Bug fix from merging Import FindCommand

ChangGittyHub committed:
Add a GUI to Duke

ChangGittyHub committed:
Improve code quality.

These are the edits I made,
*Edit Javadoc.Ensure every Javaline starts with an asterisk.
*Shorten lines with more than 150 characters.
*Add meaningful names to task objects in my Parser class.
*Change 'if' to 'else if' in Parser
*Add new line to the end of the file.
*Remove redundant code from old version.

ChangGittyHub committed:
Merge pull request #2 from ChangGittyHub/branch-A-CodeQuality

Improve code quality.

ChangGittyHub committed:
Add Assertions

Add assertions in these methods,
* add(Task task) method - test that Task parameter is not
null.
* markTaskAsDone(int index) method - test that index parameter is
within the bounds of the list.
* unmarkTask(int index) method -test that the index parameter is
within the bounds of the list.
* deleteTask(int index) method - test that the index parameter is
within the bounds of the list.
* getTask(int index) method - test that the index parameter is
within the bounds of the list.

ChangGittyHub committed:
Merge pull request #3 from ChangGittyHub/branch-A-Assertions

Add Assertions

ChangGittyHub committed:
Implement Tagging function

ChangGittyHub committed:
Merge pull request #4 from ChangGittyHub/branch-C-Tagging

Implement Tagging function

ChangGittyHub committed:
Implement a delay upon exit to show exit reply

ChangGittyHub committed:
Integrate Duuk's personality

ChangGittyHub committed:
Fix minor UI issue

ChangGittyHub committed:
Add a User Guide
